src/libcollections/btree/map.rs

```diff
@@ -342,6 +342,14 @@ pub struct OccupiedEntry<'a, K: 'a, V: 'a> {
     _marker: PhantomData<&'a mut (K, V)>,
 }
 
+// An iterator for merging two sorted sequences into one
+struct MergeIter<K, V, I: Iterator<Item=(K, V)>> {
```

Comment:
As a generally useful concept, it seems weird that `MergeIter` would be in `map.rs`. That said, I can't really think of anywhere else that it might go, so I don't know if anything should be changed here.